ESMF/GOCART2G_GridComp/DU2G_GridComp/DU2G_GridCompMod.F90

```diff
@@ -1308,14 +1301,16 @@ subroutine monochromatic_aerosol_optics(state, rc)
     call c_f_pointer(address, self)
 
     do n = 1, nbins
-      do i = 1, i2
-        do j = 1, j2
-          do k = 1, km
-            call Chem_MieQuery(self%diag_MieTable(instance), n, mieTable_index, q_4d(i,j,k,n), rh(i,j,k), tau=tau(i,j,k), __RC__)
-            tau_s(i,j,k) = tau_s(i,j,k) + tau(i,j,k)
-          end do
-        end do
-      end do
+      self%diag_Mie(instance)%Querry(n, mieTable_index, q_4d(:,:,:,n), rh, tau=tau, __RC__)
+      tau_s = tau_s + tau
+      !do i = 1, i2
+      !  do j = 1, j2
+      !    do k = 1, km
+      !      call Chem_MieQuery(self%diag_MieTable(instance), n, mieTable_index, q_4d(i,j,k,n), rh(i,j,k), tau=tau(i,j,k), __RC__)
+      !      tau_s(i,j,k) = tau_s(i,j,k) + tau(i,j,k)
+      !    end do
+      !  end do
+      !end do
```

Comment:
```suggestion
```
Note to @amdasilva - we still need to go back and verify that ELEMENTAL is providing the expected performance.